cairo/scripts/prove_block.py

```diff
@@ -0,0 +1,280 @@
+        transactions=tuple(
+            (
+                LegacyTransaction(
+                    nonce=hex_to_u256(tx["nonce"]),
+                    gas_price=hex_to_uint(tx["gasPrice"]),
+                    gas=hex_to_uint(tx["gas"]),
+                    to=Address(hex_to_bytes(tx["to"])) if tx["to"] else Bytes0(),
+                    value=hex_to_u256(tx["value"]),
+                    data=Bytes(hex_to_bytes(tx["data"])),
+                    v=hex_to_u256(tx["v"]),
+                    r=hex_to_u256(tx["r"]),
+                    s=hex_to_u256(tx["s"]),
+                )
+                if isinstance(tx, dict)
+                else Bytes(hex_to_bytes(tx))
+            )
+            for tx in fixture["newBlockParameters"]["transactions"]
+        ),
+        ommers=(),
+        withdrawals=tuple(
+            Withdrawal(
+                index=U64(int(w["index"], 16)),
+                validator_index=U64(int(w["validatorIndex"], 16)),
+                address=Address(hex_to_bytes(w["address"])),
+                amount=U256(int(w["amount"], 16)),
+            )
+            for w in fixture["newBlockParameters"]["withdrawals"]
+        ),
+    )
+
+    # Convert ancestors
+    blocks = [
+        Block(
+            header=load.json_to_header(ancestor),
+            transactions=(),
+            ommers=(),
+            withdrawals=(),
+        )
+        for ancestor in fixture["ancestors"]
+    ]
+
+    # Create blockchain
+    chain = BlockChain(
+        blocks=blocks,
+        state=convert_defaultdict(load.json_to_state(fixture["pre"])),
+        chain_id=U64(fixture["chainId"]),
+    )
+
+    # TODO: Remove when partial MPT is implemented
+    state_root = apply_body(
+        chain.state,
+        get_last_256_block_hashes(chain),
+        block.header.coinbase,
+        block.header.number,
+        block.header.base_fee_per_gas,
+        block.header.gas_limit,
+        block.header.timestamp,
+        block.header.prev_randao,
+        block.transactions,
+        chain.chain_id,
+        block.withdrawals,
+        block.header.parent_beacon_block_root,
+        calculate_excess_blob_gas(chain.blocks[-1].header),
+    ).state_root
+
+    # Recreate block with computed state root
+    block = Block(
+        header=load.json_to_header(
+            {
+                **fixture["newBlockParameters"]["blockHeader"],
+                "stateRoot": "0x" + state_root.hex(),
+            }
+        ),
+        transactions=block.transactions,
+        ommers=(),
+        withdrawals=block.withdrawals,
+    )
+    chain = BlockChain(
+        blocks=blocks,
+        state=convert_defaultdict(load.json_to_state(fixture["pre"])),
+        chain_id=U64(fixture["chainId"]),
+    )
+
+    # Prepare inputs
+    public_inputs = {
+        "block": block,
+        "blockchain": chain,
+        "block_hash": Bytes32(
+            bytes.fromhex(fixture["newBlockHash"].removeprefix("0x"))
+        ),
```

Comment:
I don't understand why we need this

Reply:
the new block hash is not part of the block object